client/my-sites/site-settings/form-general.jsx

```diff
@@ -680,33 +762,44 @@ const FormGeneral = React.createClass( {
 						{ this.syncNonPublicPostTypes() }
 
 						<CompactCard href={ '../security/' + site.slug }>
-							{ this.translate( 'View Jetpack Monitor Settings' ) }
+							{ translate( 'View Jetpack Monitor Settings' ) }
 						</CompactCard>
 						<CompactCard href={ 'https://wordpress.com/manage/' + site.ID }>
-							{ this.translate( 'Migrate followers from another WordPress.com blog' ) }
+							{ translate( 'Migrate followers from another WordPress.com blog' ) }
 						</CompactCard>
 					</div>
 					: null }
 			</div>
 		);
-	},
+	}
 
 	_showWarning( site ) {
+		const { translate } = this.props;
 		if ( ! site || ! site.options ) {
 			return;
 		}
 		if ( site.jetpack && ! site.hasMinimumJetpackVersion ) {
 			notices.warning(
-				this.translate( 'Jetpack %(version)s is required to manage Settings', {
+				translate( 'Jetpack %(version)s is required to manage Settings', {
 					args: { version: config( 'jetpack_min_version' ) }
 				} ),
 				{
-					button: this.translate( 'Update now' ),
+					button: translate( 'Update now' ),
 					href: site.options.admin_url + 'plugins.php?plugin_status=upgrade'
 				}
 			);
 		}
 	}
-} );
-
-export default protectForm( FormGeneral );
+}
+
+export default connect(
+	( state, { site } ) => {
```

Comment:
Should we use the `getSelectedSiteId` selector here instead of having the site object be passed in? Would save a lot of ternaries.